### ISSUE TITLE
services: Accessibility isssues

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#15196]
- [x] #services-dropdown is a Cockpit widget (not a PF4 one) and has an issue where it does not have an accessible name. That is, it's lacking a label. This can be fixed by adding a `<label>`, aria-label, or aria-labelledby (pointing to an ID). I'd suggest calling it "State" and consider using [the screen reader only class (`pf-u-screen-reader`)](https://www.patternfly.org/v4/utilities/accessibility#screen-reader-only) on a label.
- [ ] Cockpit logs have inappropriate usage of role="row" (see #15191, where this might be fixed instead)
- [ ] Description list has immediately children not of type `<dl>` or `<dt>`
- [x] List items are not used properly. `<li>` need to be children of a list item, such as  `<ul>` (or `<ol>`, which doesn't really apply here0. They cannot be free floating.

**Comment by KKoukiou** [comment]
>     * [ ]  #services-dropdown is a Cockpit widget (not a PF4 one) and has an issue where it does not have an accessible name. That is, it's lacking a label. This can be fixed by adding a `<label>`, aria-label, or aria-labelledby (pointing to an ID). I'd suggest calling it "State" and consider using [the screen reader only class (`pf-u-screen-reader`)](https://www.patternfly.org/v4/utilities/accessibility#screen-reader-only) on a label.
> 
>     * [ ]  Cockpit logs have inappropriate usage of role="row" (see #15191, where this might be fixed instead)
> 
>     * [ ]  Description list has immediately children not of type `<dl>` or `<dt>`

I guess you are talking about the service details here. We use PF4 Description list, https://www.patternfly.org/v4/components/description-list which has indeed `<div>` as children of the `<dl>` and according to the docs this is fine. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt
> 
>     * [ ]  List items are not used properly. `<li>` need to be children of a list item, such as  `<ul>` (or `<ol>`, which doesn't really apply here0. They cannot be free floating.

If you are talking about the services list - I can't verify what you say. All services rows are `<li>` and direct children of an `<ul>` item. Do you mean some other part of the page?

**Comment by marusak** [comment]
>  List items are not used properly. `<li>` need to be children of a list item, such as  `<ul>` (or `<ol>`, which doesn't really apply here0. They cannot be free floating.

This is in service details, for dependant services like the ones that are required...
